troubleshooting apiId error when configuring autodiscovery (#132)
mulesoft/docs-api-manager: modules/ROOT/pages/troubleshooting-config-properties-apiid-error.adoc

> = Setting API ID in Configuration File for API Autodiscovery Does Not Work

Reviewer: suggest "in Configuration" -> "in a Configuration"  or similar

Reviewer: suggest something similar to "Setting an API ID" 

I realize that makes the title even longer. 

That said, ignore my comments if the original title above is an actual error message.

Reviewer 2: It looks like the error message is actually "Failed to deploy artifact [accounts]"
Are there peer topics, and do they use the error text as title?
Suggest ensuring parallelism among peer topic titles.
It might be clearest to use the error message as the title: Failed to deploy artifact [accounts] Error Handling
And if an error message is long, we could use just the beginning of it and include ellipses: Failed to deploy... Error Handling.
Does that help?

Author: Removed Configuration File. Too many articles in the heading makes things long and convoluted.

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> When configuring the API Gateway for API autodiscovery, you might recieve the following error if you attempt to set the *apiId* property in the configuration file:

Reviewer: *apiId*: If it appears that way in the GUI, keep the bold. If it's a property you set in a file, we use `apiId`.

[source, text, linenums] 
----
org.mule.runtime.deployment.model.api.DeploymentException: Failed to deploy artifact [accounts]
Caused by: org.mule.runtime.api.exception.MuleRuntimeException: org.mule.runtime.deployment.model.api.DeploymentInitException: PropertyNotFoundException: Couldn't find configuration property 
value for key ${apiId} from properties provider Domain properties - within 
resolver Deployment properties trying to process key apiId - within 
esolver global-properties configuration attributes trying to process key apiId - within resolver environment properties provider trying to process key apiId - within resolver Deployment properties trying to process key apiId - within resolver External files trying to process key apiId - within resolver Deployment properties trying to process key apiId
----

== Solution

> This issue occurs if you did not add the *apiId* property in the config.properties file, and failed to modify the value for the *apiId* attribute in the *api-gateway:autodiscovery* element. The Autodiscovery process runs on a specific instance of the API Manager. For that reason, you must specify a valid apiId of type long.

Reviewer: backticks around config.properties (`config.properties`)

Reviewer: If *api-gateway:autodiscovery* is an XML element (and not something that appears in the GUI), we use backticks `api-gateway:autodiscovery`.

Reviewer: I think we avoid "the" before a product name: "of API Manager"
Kim would be able to say definitively.

Reviewer: *apiId* -> `apiId` for both instances of the term?

Reviewer: I'd do "type `long`"  (adding the backticks) so that translators know it's a code-like term

== To resolve this issue

> . Go to the config.properties file and append the following entry: 

Reviewer 3: What happens if the app does not have a config.properties file or it is called differently. Should we link this to how to set up a config properties file?

Author: I think that would be great. I'm not sure if there's a topic for that already. Let me check.

Reviewer: `config.properties`

+
[source, text, linenums]
----
apiId=[your-specific-value]
---- 
+

> . In your <api-gateway:autodiscovery> element, set the apiID attribute to the apiId property in the config.properties file. 

Reviewer: `<api-gateway:autodiscovery>` element

I'm not sure if we have an agreed-upon style for elements. You might want to make them match on the page. Earlier it's `api-gateway:autodiscovery` element (without the angle brackets)

= Setting API ID in Configuration File for API Autodiscovery Does Not Work
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> When configuring the API Gateway for API autodiscovery, you might recieve the following error if you attempt to set the *apiId* property in the configuration file:

Reviewer 4: There is a typo in `recieve`

Author: Fixed!

> When configuring the API Gateway for API autodiscovery, you might recieve the following error if you attempt to set the *apiId* property in the configuration file:

Reviewer 2: ```suggestion
When you configure the API gateway for API autodiscovery, you might receive the following error if you attempt to set the `apiId` property in the configuration file:
```

Author: Done!

[source, text, linenums] 
----
org.mule.runtime.deployment.model.api.DeploymentException: Failed to deploy artifact [accounts]
Caused by: org.mule.runtime.api.exception.MuleRuntimeException: org.mule.runtime.deployment.model.api.DeploymentInitException: PropertyNotFoundException: Couldn't find configuration property 
value for key ${apiId} from properties provider Domain properties - within 
resolver Deployment properties trying to process key apiId - within 
esolver global-properties configuration attributes trying to process key apiId - within resolver environment properties provider trying to process key apiId - within resolver Deployment properties trying to process key apiId - within resolver External files trying to process key apiId - within resolver Deployment properties trying to process key apiId
----

== Solution

> This issue occurs if you did not add the *apiId* property in the config.properties file, and failed to modify the value for the *apiId* attribute in the *api-gateway:autodiscovery* element. The Autodiscovery process runs on a specific instance of the API Manager. For that reason, you must specify a valid apiId of type long.

Reviewer 4: @priyaayyar @balbifm should we add a note somewhere stating that if you deploy through API Manager or download the auto-generated proxy this configuration is automatically setup?

Author: Yes, I think we should.

Author: Fixed!

> This issue occurs if you did not add the *apiId* property in the config.properties file, and failed to modify the value for the *apiId* attribute in the *api-gateway:autodiscovery* element. The Autodiscovery process runs on a specific instance of the API Manager. For that reason, you must specify a valid apiId of type long.

Reviewer 2: ```suggestion
This issue occurs if you did not add the `apiId` property in the `config.properties` file, and failed to modify the value for the `apiId` attribute in the `api-gateway:autodiscovery` element. The autodiscovery process runs on a specific instance of Anypoint API Manager. For that reason, you must specify a valid `apiId`value of type `long`.
```

Author: Done

> == To resolve this issue

Reviewer 2: This heading repeats "Solution"...which makes me realize that most of the previous text is NOT a solution but the CAUSE of the error. It would be great if we could template our troubleshooting topics to...:

**Oh My Gosh You Messed Up Error Handling** (title)

When you do this thing to accomplish this goal, you sometimes see this thing happen and the message Oh My Gosh You Messed Up appears.

**Cause**
This is caused because you messed up by doing this or not doing that.

**Solution**
Go back and fix it:
1. From this location, click this.
2. Enter that.
3. Click Save.

Author: Make sense! Done!

> . Go to the config.properties file and append the following entry: 

Reviewer 2: This wording is not clear, at least to me. "append the following entry" sounds like it should be "append <this value> to the <xyz> entry/parameter/value/whatever it is.

Clarify 1) what is already there, and 2) what you are appending to the end of that already-there thing

Author: Added "append the following property and value pair"

+
[source, text, linenums]
----
apiId=[your-specific-value]
----
+

> . In your <api-gateway:autodiscovery> element, set the apiID attribute to the apiId property in the config.properties file. 

Reviewer 2: ```suggestion
. In your `<api-gateway:autodiscovery>` element, set the `apiID` attribute to the `apiId` property in the `config.properties` file. 
```

Reviewer 2: Doublecheck with Duke, but I am pretty sure that we want all syntax elements to appear in monospace, which is achieved using backticks.
Boldface is for UI labels and user entry (user entry being boldface mono).
Let me know if you think I am wrong about this. I am still trying to get consensus on the use of bold.
